### PR TITLE
Allow Shape Updates by Owner or Presenter

### PR DESCRIPTION
**Pull request by KDSBrowne**
### What does this PR do?
Allows the presenter to lock their shapes so other meeting participants can't edit or unlock.
Shape updates can only be persisted if done by the owner or presenter.

### CODE REVIEW

File: bigbluebutton-html5/imports/ui/components/whiteboard/component.jsx [hunk 1]
```diff
@@ -255,32 +255,22 @@ export default function Whiteboard(props) {
     }
   }, [isPanning]);
 
+  const isOwner = (id) => {
+    if (currentUser.presenter) return true;
+    const owner = shapes[id]?.userId;
+    return (owner && (owner !== currentUser?.userId)) ? false : true;
```
[the remark this answers is not in the document]
Comment (antobinary) [reply bigbluebutton-html5/imports/ui/components/whiteboard/component.jsx:261]: You have `currentUser?.userId` but above it you have `currentUser.presenter` without `?`
Would you consider checking once in the beginning `if  (!currentUser) return false;` or something like it? I am not sure what a correct safe return value is.

[the remark this answers is not in the document]
Comment (KDSBrowne) [reply bigbluebutton-html5/imports/ui/components/whiteboard/component.jsx]: Updated the function return

File: bigbluebutton-html5/imports/ui/components/whiteboard/component.jsx [hunk 2]
```diff
@@ -256,9 +256,9 @@ export default function Whiteboard(props) {
   }, [isPanning]);
 
   const isOwner = (id) => {
-    if (currentUser.presenter) return true;
     const owner = shapes[id]?.userId;
-    return (owner && (owner !== currentUser?.userId)) ? false : true;
+    const hasShapeAccess = (owner && (owner === currentUser?.userId) || (!owner && !shapes[id]) || currentUser?.presenter);
```
Comment (antobinary) [inline bigbluebutton-html5/imports/ui/components/whiteboard/component.jsx:260]:
```suggestion
    const hasShapeAccess = (owner && (owner === currentUser?.userId)) || (!owner && !shapes[id]) || currentUser?.presenter;

```

**Comment by germanocaumo** [comment]
Replaced by https://github.com/bigbluebutton/bigbluebutton/pull/15636, closing

**Pull request closed without merging**